### PR TITLE
Fixes #277. Adds 'includes' method to DefineList

### LINKED ISSUE #277: Add includes method to can-define/list/list
As a user, I’d find it really helpful if the [includes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes) method was available on lists.

### PR DESCRIPTION
**Pull request by mjstahl**
Resolves https://github.com/canjs/can-define/issues/277

### CODE REVIEW

**Comment by mjstahl** [comment]
@justinbmeyer So this should also check equality of DefineMap's or other DefineList's?

File: list/list.js [hunk 1]
```diff
@@ -472,6 +472,25 @@ function a(fnLength, fnName) {
 });
 
 assign(DefineList.prototype, {
+	includes: function(item, fromIndex) {
+		// adapted from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes#Polyfill
```
Comment (justinbmeyer) [inline list/list.js]: `Array.prototype.includes` is not in IE11  :-(

**Comment by justinbmeyer** [comment]
I think we should make it use Array.prototype.includes.  It's not supported in IE.  We should warn if includes is called and there is no polyfill.

```js



includes: (function(){
  var arrayIncludes =  Array.protoytype.includes;
  if(arrayIncludes){
    return function includes(){
      return arrayIncludes.apply(this, arguments); // is .includes in an Observation observable
    }
  } else {
    return function includes(){
      throws new Error("DefineList.prototype.includes must have Array.prototype.includes available. Please add a polyfill to this environment.");
    }
  }
})()
```


```js
new Observation(function(){
  return list.includes("a")
})
```

File: list/list.js [hunk 2]
```diff
@@ -57,6 +57,19 @@ function getSchema() {
 	return schema;
 }
 
+var includes = (function(){
```
[the remark this answers is not in the document]
Comment (cherifGsoul) [reply list/list.js]: I changed it @justinbmeyer

**Pull request merged**